ycmd/completers/cpp/clang_completer.py

```diff
@@ -108,26 +108,72 @@ def DefinedSubcommands( self ):
              'GoToDeclaration',
              'GoTo',
              'GoToImprecise',
-             'ClearCompilationFlagCache']
+             'ClearCompilationFlagCache',
+             'GetType',
+             'GetParent']
 
 
-  def OnUserCommand( self, arguments, request_data ):
+  def OnUserCommand( self, arguments, request_data ): 
     if not arguments:
       raise ValueError( self.UserCommandsHelpMessage() )
 
-    command = arguments[ 0 ]
-    if command == 'GoToDefinition':
-      return self._GoToDefinition( request_data )
-    elif command == 'GoToDeclaration':
-      return self._GoToDeclaration( request_data )
-    elif command == 'GoTo':
-      return self._GoTo( request_data )
-    elif command == 'GoToImprecise':
-      return self._GoToImprecise( request_data )
-    elif command == 'ClearCompilationFlagCache':
-      return self._ClearCompilationFlagCache()
-    raise ValueError( self.UserCommandsHelpMessage() )
-
+    #
+    # An if/elif for each of the options fails the cyclomatic complexity check
+    # in the test suite. python doesn't have a switch statement, and
+    # recommendations typically point to using a dictionary. So we use a
+    # dictionary to map the entered command to the method which implements it
+    # via some (arguably ugly) metaprogramming.
```

Comment:
I don't think you need the above paragraph. The bottom part of the comment is fine.